### PR TITLE
Expose multiple ports on ssh

### LINKED ISSUE #211: Remove/fix security rule deletion
https://github.com/citusdata/test-automation/blob/master/hammerdb/connect-driver.sh deletes a security rule, but that's against Azure policy.

### PR DESCRIPTION
**Pull request by SaitTalhaNisanci**
Expose multiple ports on ssh so that we don't hit security group rules when trying to connect via the default port, 22.

The custom port we now use for ssh is 3456.

Alternatively I have tried to use a loadbalancer to have port forwarding but it seems that in that case we again hit rule 103 because only internal requests from loadbalancer to vms are allowed and public requests are not.

Fixes #211.

### CODE REVIEW

File: azure/azuredeploy.json
Comment (SaitTalhaNisanci) [inline azure/azuredeploy.json:404]: Will update this to master before merging

File: README.md
Comment (onurctirtir) [inline README.md:108]: instead of removing this line, maybe we should provide port 3456 too

File: README.md
Comment (onurctirtir) [inline README.md:179]: same

File: azure/create-cluster.sh
Comment (onurctirtir) [inline azure/create-cluster.sh:65]:
```suggestion
echo "and connect with: ${connection_string}"
```

File: azure/connect.sh
Comment (onurctirtir) [inline azure/connect.sh:22]: minor: maybe one more newline as we did previously

File: azure/init.sh
Comment (onurctirtir) [inline azure/init.sh:50]: why putting port 22 ?

File: hammerdb/azuredeploy.json
Comment (onurctirtir) [inline hammerdb/azuredeploy.json:254]: this might deserve a comment like:

add security rule to enable custom ssh port

File: hammerdb/create-run.sh
Comment (onurctirtir) [inline hammerdb/create-run.sh:26]: can we somehow deduce custom ssh port instead of hardcoding it (if it wouldn't require any additional efforts)?

File: hammerdb/driver-init.sh
Comment (onurctirtir) [inline hammerdb/driver-init.sh:46]: again, why putting 22 ?

File: azure/init.sh
Comment (onurctirtir) [reply azure/init.sh:50]: I guess it's because now we override ssh ports for custom port and so the default port would go away if we don't add it explicitly

File: azure/init.sh
Comment (onurctirtir) [inline azure/init.sh:54]: [ let me do some research for what is semanage :) ]

File: README.md
Comment (SaitTalhaNisanci) [reply README.md:108]: Do you think it is valuable? `./connect.sh` seems enough. There wasn't any connect script before and there was only this alternative in the past, so now using `connect.sh` is what we do

File: azure/init.sh
Comment (SaitTalhaNisanci) [reply azure/init.sh:50]: I kept the default port because Azure has rules based on rule 22 and it tries to connect on 22. So I thought it could also cause some problems and possibly creation of ICM if rule 22 is not working for inner communication in azure.

File: README.md
Comment (onurctirtir) [reply README.md:108]: true, it doesn't seem valuable

File: azure/init.sh
Comment (SaitTalhaNisanci) [reply azure/init.sh:54]: we have secure linux and in secure linux, we need to do some extra commands to enable some security related stuff. semanage helps with adding the port in this case

File: hammerdb/azuredeploy.json
Comment (SaitTalhaNisanci) [reply hammerdb/azuredeploy.json:254]: You mean a separate comment other than the current description 2 lines below or update the description below?

File: azure/create-cluster.sh
Comment (onderkalaci) [reply azure/create-cluster.sh:65]: I prefer `echo ${connection_string}` as it makes it easy to copy & paste.

> ALTERNATIVELY:

Maybe `ALTERNATIVELY RUN THE FOLLOWING:`

File: hammerdb/azuredeploy.json
Comment (onderkalaci) [inline hammerdb/azuredeploy.json:263]: what happens if there is already a rule with `"priority": 102,`? Or, could that happen?

Comment (SaitTalhaNisanci) [reply hammerdb/azuredeploy.json:263]: there is already a rule with 102 :)

It gets appended after all the existing rules. So 100, 101, 102, and 103 will be reserved. and these rules will have priority of 104, 105...

File: hammerdb/azuredeploy.json
Comment (onurctirtir) [reply hammerdb/azuredeploy.json:254]: > update the description below?

makes more sense, maybe:

"description": "allow accessing to custom ssh port from all ip's",

**Comment by onderkalaci** [comment]
btw, I tested this and works as it used to be, thanks

**Review by onurctirtir — APPROVED** [review]
let's not forget to revert those before merging

File: azure/create-cluster.sh
Comment (SaitTalhaNisanci) [reply azure/create-cluster.sh:65]: @onurctirtir FYI I changed this as `ALTERNATIVELY RUN THE FOLLOWING:` as it can indeed easier for copy & paste, I thought it was okay for you since you approved, let me know if it is not

**Comment by SaitTalhaNisanci** [comment]
Thanks for the quick review @onurctirtir.

**Pull request merged**
